src/jit/lsra.cpp

```diff
@@ -1930,8 +1937,10 @@ void LinearScan::identifyCandidatesExceptionDataflow()
 
 bool LinearScan::isRegCandidate(LclVarDsc* varDsc)
 {
-    // Check to see if opt settings permit register variables
-    if ((compiler->opts.compFlags & CLFLG_REGVAR) == 0)
+    // We shouldn't be called if opt settings do not permit register variables.
+    assert((compiler->opts.compFlags & CLFLG_REGVAR) != 0);
+
+    if (!varDsc->lvTracked)
     {
```

Comment:
Unrelated: it seems like this code from identifyCandidates() should be here:
```
#if !defined(_TARGET_64BIT_)
        if (varDsc->lvType == TYP_LONG)
        {
            // Long variables should not be register candidates.
            // Lowering will have split any candidate lclVars into lo/hi vars.
            varDsc->lvLRACandidate = 0;
            continue;
        }
#endif // !defined(_TARGET_64BIT)
```
as:
```
#if !defined(_TARGET_64BIT_)
        if (varDsc->lvType == TYP_LONG)
        {
            // Long variables should not be register candidates.
            // Lowering will have split any candidate lclVars into lo/hi vars.
            return false;
        }
#endif // !defined(_TARGET_64BIT)
```